misc/python/materialize/mzbuild.py

```diff
@@ -160,6 +155,10 @@ def inputs(self) -> Set[str]:
         """Return the files which are considered inputs to the action."""
         raise NotImplementedError
 
+    def extra(self) -> str:
```

Comment:
Maybe instead of `extra`, this should be `build_params` and return a dictionary?

Reply:
Makes sense in principle, but on first glance I'm not 100% sure how to reliably incorporate a dictionary into the fingerprint. Is calling `str` sufficient? I know Python's guarantees on dict ordering have changed over the years. So going to leave as a `str` for now because I am more convinced of correctness, but would certainly be happy if someone else wanted to the legwork to clean this up later!